### PR TITLE
WOL-76

### PR DESCRIPTION
**Pull request by kangkanbharadwaj**
Added a cpp inference support for TRT engine.

So just inference takes 3-4 ms, combine with pre processing and post, it takes 50-60ms. Doc strings are added.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.10)
```
Comment (ahoereth) [inline CMakeLists.txt:1]: Don't you want to have a cpp subdirectory? I feel like the top level is becoming very convoluted.

File: pure_trt.cpp [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+#include <opencv2/highgui.hpp>
+#include <opencv2/core.hpp>
+#include <algorithm>
+#include <numeric>
+#include <cstdlib>
+#include <sstream>
+#include "logger.h"
+
+// destroy TensorRT objects if something goes wrong
+struct TRTDestroy
```
Comment (ahoereth) [inline pure_trt.cpp:21]: Should we just right away have proper doxygen comments? Including a file-level docstring of what this is doing?

File: pure_trt.cpp [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        std::cerr << "usage: " << argv[0] << " model.onnx image.jpg wolffkran_classes.txt\n";
+        return -1;
+    }
+    std::string model_path(argv[1]);
+    std::string image_path(argv[2]);
+    std::string classnames_path(argv[3]);
+    int batch_size = 1;
+
+    // initialize TensorRT engine and parse ONNX model
+    TRTUniquePtr<nvinfer1::ICudaEngine> engine = loadEngine(model_path);
```
Comment (ahoereth) [inline pure_trt.cpp:178]: Are you loading an *.engine (tensorrt) or an *.onnx (onnx) file?

**Review by ahoereth — COMMENTED** [review]
Cool! Did you run it a couple thousand times already to get an idea of how long the inference step takes?

File: CMakeLists.txt (hunk 1 above)
Comment (kangkanbharadwaj) [reply CMakeLists.txt:1]: yes I am doing that right now!! :) That's the structure in the jetson with a subdir

File: pure_trt.cpp (hunk 2 above)
Comment (kangkanbharadwaj) [reply pure_trt.cpp:21]: yes we should. Actually there are some stuff that myself is trying ti understand what is going on

**Comment by ahoereth** [comment]
Also: Please have proper PR titles maybe? Small things like these should be possible

File: pure_trt.cpp (hunk 3 above)
Comment (kangkanbharadwaj) [reply pure_trt.cpp:178]: loading .engine. Those are just for name sake. May be I change that. Its misleading. 
this is what I pass to the exe: ./pure_trt ../../models/320-tiny-nothresh-nocombined-ops11.engine ../../images/000000000202.jpg ../wolffkran_classes.txt

**Comment by kangkanbharadwaj** [comment]
> Cool! Did you run it a couple thousand times already to get an idea of how long the inference step takes?

Not yet. I figured out an error and I am trying to fix that. Happens when trying to copy image data to a pointer

File: pure_trt.cpp (hunk 3 above)
Comment (ahoereth) [reply pure_trt.cpp:178]: well yea, please have the code represent that. Its not an ONNX model anymore if its an *.engine file! Very different.

Comment (kangkanbharadwaj) [reply pure_trt.cpp]: done

**Comment by ahoereth** [comment]
is this done and ready for review?

**Comment by ahoereth** [comment]
@kangkanbharadwaj why was this merged without review? You pushed a whole lot of binaries and the build directory to master, making everybody download them forever in the future.

**Pull request merged**